### DISCUSSION TITLE
Yourls not working with PHP 8.1

### DISCUSSION THREAD

**Discussion by DrZen33** [YOURLS/YOURLS#3165] [Q&A] [answered]
Hello
I have Yourls 1.8.2 installed and it has been working 100% OK with PHP 7.4 on the server.
I have multiple websites on same server and needed to upgrade to latest PHP version.
When I switched the server to PHP 8.1, Yourls stopped working altogether and went to an error page.
I've had to return PHP back to version 7.4.

Is Yourls compatible with PHP 8.1 ?
Is there an extra setting/extension in PHP 8.* that needs to be turned on, to make Yourls work ?

Regards
Derek

**Comment by dgw** [comment]
If there's an error page, there's almost certainly an error in a log somewhere. Follow [troubleshooting procedures](https://github.com/YOURLS/YOURLS/wiki/Troubleshooting-first-steps) to find it.

You didn't say what kind of server (Apache, Nginx, IIS, Caddy, etc.) this is. Different servers talk to PHP in different ways, and it's possible that that's the problem. (E.g. it's somewhat common to set up Nginx with PHP-FPM over a Unix socket, the path to which changes depending on the PHP version installed.)

**Comment by DrZen33** [comment, marked as answer]
Hi dgw
Thank you very much for your super fast reply!!!
Server is Litespeed.
I checked the error log file inside of the Admin folder for Yourls.
It reported the following:
PHP Fatal error:  Uncaught Error: Class "PDO" not found in /home/******/zen3.me/includes/vendor/aura/sql/src/AbstractExtendedPdo.php:26

I then went to the Extensions for PHP 8.1 and turned on all of these:
pdo
pdo_dblib
pdo_mysql

Yourls is now functioning 100% OK using PHP 8.1
Case closed.

![php-81-pdo](https://user-images.githubusercontent.com/20509046/146625031-87785092-d161-479a-a9bc-9b813148ac8d.png)
.

**Comment by ozh** [comment]
It seems that YOURLS is [working fine](https://test.yourls.org/altl) on [8.1](https://test.yourls.org/) so the problem is elsewhere (most likely, bad editing of a file, or files incorrectly uploaded to server)

Edit : just read your answer -- all is fine :)
